source/patterns/@aws-solutions-constructs/aws-fargate-sqs/test/fargate-sqs.test.ts

```diff
@@ -0,0 +1,379 @@
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import '@aws-cdk/assert/jest';
+import * as defaults from '@aws-solutions-constructs/core';
+import * as cdk from "@aws-cdk/core";
+import { FargateToSqs } from "../lib";
+import * as sqs from '@aws-cdk/aws-sqs';
+import * as ecs from '@aws-cdk/aws-ecs';
+
+test('New service/new queue, dlq, public API, new VPC', () => {
+
+  // An environment with region is required to enable logging on an ALB
+  const stack = new cdk.Stack(undefined, undefined, {
+    env: { account: "123456789012", region: 'us-east-1' },
+  });
+  const publicApi = true;
+  const clusterName = "custom-cluster-name";
+  const containerName = "custom-container-name";
+  const serviceName = "custom-service-name";
+  const queueName = "custom-queue-name";
+
+  new FargateToSqs(stack, 'test-construct', {
+    publicApi,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+    vpcProps: { cidr: '172.0.0.0/16' },
+    clusterProps: { clusterName },
+    containerDefinitionProps: { containerName },
+    fargateTaskDefinitionProps: { family: 'family-name' },
+    fargateServiceProps: { serviceName },
+    queueProps: { queueName },
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Service", {
+    LaunchType: 'FARGATE',
+    DesiredCount: 2,
+    DeploymentConfiguration: {
+      MaximumPercent: 150,
+      MinimumHealthyPercent: 75
+    },
+    PlatformVersion: ecs.FargatePlatformVersion.LATEST,
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Service", {
+    ServiceName: serviceName
+  });
+  expect(stack).toHaveResourceLike("AWS::ECS::TaskDefinition", {
+    Family: 'family-name'
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Cluster", {
+    ClusterName: clusterName
+  });
+
+  expect(stack).toHaveResourceLike("AWS::SQS::Queue", {
+    QueueName: queueName
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::TaskDefinition", {
+    ContainerDefinitions: [
+      {
+        Essential: true,
+        Image: {
+          "Fn::Join": [
+            "",
+            [
+              "123456789012.dkr.ecr.us-east-1.",
+              {
+                Ref: "AWS::URLSuffix"
+              },
+              "/fake-repo:latest"
+            ]
+          ]
+        },
+        MemoryReservation: 512,
+        Name: containerName,
+        PortMappings: [
+          {
+            ContainerPort: 8080,
+            Protocol: "tcp"
+          }
+        ]
+      }
+    ]
+  });
+
+  expect(stack).toHaveResourceLike("AWS::EC2::VPC", {
+    CidrBlock: '172.0.0.0/16'
+  });
+  // Confirm we created a Public/Private VPC
+  expect(stack).toHaveResourceLike('AWS::EC2::InternetGateway', {});
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::SQS::Queue', 2);
```

Comment:
We should have one test that specifies DeadLetterQueue props and checks that a queue with the proper name exists. Looking at all our other SQS constructs, it looks like this is something we have missed completely up to this point.